lib/chef/chef_fs/file_system/repository/directory.rb

```diff
@@ -149,6 +150,10 @@ def exists?
 
           protected
 
+          def write(data)
+            raise "called write on #{file_path}"
```

Comment:
oof, that was debug spew that shouldn't have stayed in. But now you mention it, I think it's reasonable to raise a `Chef::ChefFS::FileSystem::FileSystemError` if someone does try to write to a directory entry.